### PR TITLE
fix(core): fix upload file guard

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Fix file upload API.

The `koa-body` has been upgraded to the latest version, which caused the file upload API to break. This change fixes the issue.

The `ctx.request.files.file` in the new version is an array, so the code has been updated to pick the first one.

## Testing

Local tested.

## Checklist

- [x] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/user-assets.ts [hunk 1]
```diff
@@ -52,13 +52,15 @@ export default function userAssetsRoutes<T extends ManagementApiRouter>(
     '/user-assets',
     koaGuard({
       files: object({
-        file: uploadFileGuard,
+        file: uploadFileGuard.array(),
```
Comment (simeng-li) [inline packages/core/src/routes/user-assets.ts:55]: .array().min(1) ?

File: packages/core/src/routes/user-assets.ts [hunk 2]
```diff
@@ -52,13 +52,15 @@ export default function userAssetsRoutes<T extends ManagementApiRouter>(
...
       }),
       response: userAssetsGuard,
     }),
     async (ctx, next) => {
-      const { file } = ctx.guard.files;
+      const { file: bodyFiles } = ctx.guard.files;
 
+      const file = bodyFiles[0];
+      assertThat(file, 'guard.file_not_found');
```
Comment (simeng-li) [inline packages/core/src/routes/user-assets.ts:63]: how about, letting the zod to guard the file length? Just throw guard.invalid_input here as a callback error

**Review by gao-sun — APPROVED** [review]
should we add some integration tests for this?

**Review by charIeszhao — APPROVED** [review]

File: packages/core/src/routes/user-assets.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/user-assets.ts:63]: This is still needed to make TS happy

File: packages/core/src/routes/user-assets.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/user-assets.ts:55]: updated

File: packages/core/src/routes/user-assets.ts (hunk 2 above)
Comment (wangsijie) [reply packages/core/src/routes/user-assets.ts:63]: updated

**Pull request merged**